src/ui.js

```diff
@@ -302,6 +302,52 @@ const ui = {
     //console.log("un_hookup_ui", elem.id)
   },
 
+  injectSvgNode: (node, targetNode, width, height = width) => {
+    node.setAttribute('x', 0)
+    node.setAttribute('y', 0)
+    const originalWidth = node.getAttribute('width')
+    const originalHeight = node.getAttribute('height')
+
+    targetNode.setAttribute('width', width)
+    targetNode.setAttribute('height', height)
+    targetNode.setAttribute('viewBox', `0 0 ${originalWidth} ${originalHeight}`)
+    targetNode.appendChild(node)
+  },
+
+  generateActivityItemFromTemplate: (playerAction, beforeNode, afterNode) => {
+    const templateActivityItem = document.querySelector('#template_activity_item')
+
+    templateActivityItem.content.querySelector('.activity_item_timestamp').innerText = new Date().toLocaleTimeString()
+
+    const playerNameSpan = templateActivityItem.content.querySelector('#activity_item_player_name')
+    playerNameSpan.innerText = localStorage.getItem('profile_name') || 'Unknown'
+    playerNameSpan.style.color = localStorage.getItem('profile_color')
+
+    templateActivityItem.content.querySelector('#activity_item_player_action').innerText = playerAction
+
+    const beforeNodeWrapper = templateActivityItem.content.querySelector('#activity_item_before_node')
+    const afterNodeWrapper = templateActivityItem.content.querySelector('#activity_item_after_node')
+
+    const size = 64
+    ui.injectSvgNode(beforeNode, beforeNodeWrapper, size)
+    ui.injectSvgNode(afterNode, afterNodeWrapper, size)
+
+    return document.importNode(templateActivityItem.content, true)
```

Comment:
I think it was clearer (at least, to me) when you used `insertBefore` like your previous code.  I think you'd need to cloneNode the templateActivityItem.  (According to MDN you're going to want to clone in anyway, even if you used importNode: https://developer.mozilla.org/en-US/docs/Web/API/Document/importNode)

But again, this may be an area where I don't know the best way?

Reply:
We're still using `insertBefore`. It's just in the next function: `addActivityLogItem` 
`importNode` performs the clone and returns it to the `addActivityLogItem` function. 
Let me know if we're on the same page.

Reply:
ah.  cool